src/configMaker.js

```diff
@@ -12,15 +15,59 @@ const makeConfig = (vars) => {
     }).join(".");
 
     const configObject = {};
-    _.set(configObject, assignPath, vars[configKey]);
+    set(configObject, assignPath, vars[configKey]);
 
     configObjectsArray.push(configObject);
     return true;
   });
 
-  return _.merge({}, ...configObjectsArray);
+  return merge({}, ...configObjectsArray);
+};
+
+const mergeConfigSections = (metadata, sections) => {
+  return Object.keys(sections).reduce(
+    (acc, key) => {
+      const section = {};
+      section[key] = sections[key](metadata);
+      return merge(acc, section);
+    },
+    metadata,
+  );
+};
+
+/* Build the full application configuration.
+ */
+const buildConfig = (name, defaults, vars, debug = false, testing = false) => {
+  const injector = getInjector();
+
+  // load environment variables
+  const metadata = new Metadata(
+    name,
+    debug,
+    testing,
+  );
+
+  const configFromEnviron = makeConfig(vars);
+
+  const environment = {
```

Comment:
This used to be called `metadata` but it wasn't metadata in the same way that metadata was in `microcosm` so I renamed it.